File: informatika_ismeretek/kozep/2020_may/c#/SudokuCLI.cs
using System;
using System.Collections.Generic;
using System.IO;

var feladvanyok = new List<Feladvany>();
var lines = File.ReadAllLines("feladvanyok.txt");

foreach(var line in lines) {
    feladvanyok.Add(new Feladvany(line));
}

Console.WriteLine($"3. Feladat: Feladványok száma: {feladvanyok.Count}");
Console.WriteLine("4. Feladat:");

var bekertMeret = MeretetBeker();
var bekertMeretuek = new List<Feladvany>();

foreach(var feladvany in feladvanyok) {
    if(feladvany.meret == bekertMeret) {
        bekertMeretuek.Add(feladvany);
    }
}

var kivalasztott = bekertMeretuek[new Random().Next(bekertMeretuek.Count)];

Console.WriteLine($"Ebből a méretből {bekertMeretuek.Count} db van");
Console.WriteLine($"5. Feladat: {kivalasztott.feladvanyTeljes}");

var kitoltottCellakSzama = 0;
for(var i = 0; i < kivalasztott.feladvanyTeljes.Length; ++i) {
    if(kivalasztott.feladvanyTeljes[i] != '0') {
        ++kitoltottCellakSzama;
    }
}

Console.WriteLine("6. Feladat: A feladvány kitöltöttsége: " + (int) ((float) kitoltottCellakSzama / kivalasztott.feladvanyTeljes.Length * 100) + "%");
Console.WriteLine("7. Feladat:");
kivalasztott.Kirajzol();

var bekertMeretuekFeladvanyai = new List<String>();
foreach(var feladvany in bekertMeretuek) {
    bekertMeretuekFeladvanyai.Add(feladvany.feladvanyTeljes);
}

File.WriteAllLines($"sudoku{bekertMeret}.txt", bekertMeretuekFeladvanyai);


int MeretetBeker() {
    while(true) {
        Console.WriteLine("Kérem 1 feladvány méretét!");
        var bekertMeret = int.Parse(Console.ReadLine());

        if(bekertMeret >= 4 && bekertMeret <= 9) {
            return bekertMeret;
        }
    }
}

File: informatika_ismeretek/kozep/2020_may/java/SudokuCLI.java
import java.io.*;
import java.nio.file.*;
import java.util.*;

public class SudokuCLI {

    public static void main(String[] args) throws IOException {
        var feladvanyok = new ArrayList<Feladvany>();

        for(var line : Files.readAllLines(Path.of("feladvanyok.txt"))) {
            feladvanyok.add(new Feladvany(line));
        }

        System.out.println("3. Feladat: Feladványok száma: " + feladvanyok.size());
        System.out.println("4. Feladat:");

        var bekertMeret = meretetBeker();
        var bekertMeretuek = new ArrayList<Feladvany>();

        for(var feladvany : feladvanyok) {
            if(feladvany.meret == bekertMeret) {
                bekertMeretuek.add(feladvany);
            }
        }

        var kivalasztott = bekertMeretuek.get(new Random().nextInt(bekertMeretuek.size()));

        System.out.println("Ebből a méretből " + bekertMeretuek.size() + " db van");
        System.out.println("5. Feladat: " + kivalasztott.feladvanyTeljes);

        var kitoltottCellakSzama = 0;
        for(var i = 0; i < kivalasztott.feladvanyTeljes.length(); ++i) {
            if(kivalasztott.feladvanyTeljes.charAt(i) != '0') {
                ++kitoltottCellakSzama;
            }
        }

        System.out.printf("6. Feladat: A feladvány kitöltöttsége: %.0f%%\n", ((float) kitoltottCellakSzama / kivalasztott.feladvanyTeljes.length() * 100));
        System.out.println("7. Feladat:");
        kivalasztott.kirajzol();

        var bekertMeretuekFeladvanyai = new ArrayList<String>();
        for(var feladvany : bekertMeretuek) {
            bekertMeretuekFeladvanyai.add(feladvany.feladvanyTeljes);
        }

        Files.write(Path.of("sudoku" + bekertMeret + ".txt"), bekertMeretuekFeladvanyai);
    }

    public static int meretetBeker() {
        while(true) {
            var bekertMeret = Integer.parseInt(System.console().readLine("Kérem 1 feladvány méretét: "));

            if(bekertMeret >= 4 && bekertMeret <= 9) {
                return bekertMeret;
            }
        }
    }
}